snapshotter/devmapper/pool_device.go

```diff
@@ -264,22 +305,34 @@ func (p *PoolDevice) DeactivateDevice(ctx context.Context, deviceName string, de
 	return nil
 }
 
+// IsActivated returns true if thin-device is activated and not suspended
+func (p *PoolDevice) IsActivated(deviceName string) bool {
+	infos, err := dmsetup.Info(deviceName)
+	if err != nil || len(infos) == 0 {
+		// Couldn't query device info, device not active
+		return false
+	}
+
+	if devInfo := infos[0]; devInfo.Suspended {
```

Comment:
Will `infos` length ever be not 1? If it is suspected to be 1, then we should return an error if `len(infos) != 1`. If the length can be any number, we should change this to be a for loop.

Reply:
Len can be either 0 (if no device) or 1, since it's not possible to have 2 devices with same name.

Reply:
So if it isn't 0, which it would be since we got to this line, we need to check that `len` equals to 1. If it does not, return an error. I just don't want to make any assumptions that it will always be 0 or 1 and at least return an error for unexpected behavior.

Reply:
Fixed